modules/apps/transfer/internal/denom/denom.go

```diff
@@ -0,0 +1,41 @@
+package denom
+
+import (
+	"strings"
+
+	"strings"
+
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+)
+
+// ExtractPathAndBaseFromFullDenom returns the trace path and the base denom from
+// the elements that constitute the complete denom.
+func ExtractPathAndBaseFromFullDenom(fullDenomItems []string) ([]string, string) {
```

Comment:
note: this was moved from trace.go, and made internal. The one difference is a pathSlice is returned instead of a path. This makes the fn work for both use cases.